sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -91,22 +91,76 @@ namespace Azure { namespace Core { namespace Http {
   };
 
   /**
-   * HTTP request method.
-   *
+   * @brief The method to be performed on the resource identified by the Request.
    */
   class HttpMethod final {
   public:
     HttpMethod() = delete;
+
+    /**
+     * @brief Constructs a new HttpMethod from a given string.
```

Comment:
Would using `HttpMethod` in the description result in a circular link to the same page on the generated docs? I remember we wanted to avoid that.

So, it might be better to split it and say `HTTP method`.

Reply:
Yes, but I think is fine, the link takes you the page with the info about an HTTPMethod

![image](https://user-images.githubusercontent.com/24213737/119565476-d733be00-bd5e-11eb-8dd9-bf6d11cb7603.png)

Reply:
![image](https://user-images.githubusercontent.com/6527137/119565995-79ec3c80-bd5f-11eb-892f-663277d92aa2.png)

Where does clicking on `HttpMethod` in that public member function section go to? To the same page?

Reply:
I think, in this case (and this is the only place I'm suggesting it, I believe), that we should say "Constructs `HttpMethod` from string", meaning, not the "HTTP method" as a broader concept, but specifically HttpMethod class instance. And we backtick it in doxygen documentation.
Plus,
* "a new" is not needed (`operator new` does the allocation, constructor does initialize. Factory method does both, but in this case it is a constructor and not a factory method). And, "a new", in the sense it was meant (i.e. casual and not C++-specific), is redundant, because there is not much other options.
* "from a given", sounds redundant. "Constructs X from Y" is a cleaner declarative-style description. And, "a given" is also redundant, because there is no much other options.

Reply:
> and we backtick it in doxygen documentation.

Backtick to remove the link but keep it formatted as a "type", differentiated from the rest of the text, like `HttpMethod`?

I like that idea. It follows the spirit of the docs issue I shared above, to avoid unnecessary clickable links that aren't helpful.

Reply:
Exactly - we've done it in other places already, Rick did it in his docs PR. So, when you need to refer to your own class name within the class documentation, we should use it to avoid self-referencing.

Reply:
backtick it in doxygen documentation does not remove the link

Reply:
Then, we should use `%`: https://www.doxygen.nl/manual/autolink.html#linkclass